### PR TITLE
Model

### PR DESCRIPTION
**Pull request by stpenney**

### CODE REVIEW

File: WebCrawler/MainWindow.xaml [hunk 1]
```diff
@@ -7,6 +7,14 @@
...
-        
+        <DataGrid x:Name="dataGridView1" HorizontalAlignment="Left" Height="243" Margin="317,102,0,0" VerticalAlignment="Top" Width="423"/>
+        <Label Content="Crawl Log" HorizontalAlignment="Left" Margin="317,76,0,0" VerticalAlignment="Top"/>
+        <Label Content="Starting Website" HorizontalAlignment="Left" Margin="76,76,0,0" VerticalAlignment="Top"/>
+        <Label Content="Topic Keywords" HorizontalAlignment="Left" Margin="76,188,0,0" VerticalAlignment="Top"/>
+        <TextBox x:Name="textBox1" HorizontalAlignment="Left" Height="23" Margin="76,102,0,0" TextWrapping="Wrap" VerticalAlignment="Top" Width="120"/>
+        <TextBox x:Name="textBox2" HorizontalAlignment="Left" Height="23" Margin="76,214,0,0" TextWrapping="Wrap" VerticalAlignment="Top" Width="120"/>
+        <Button x:Name="button1" Content="Add" HorizontalAlignment="Left" Margin="201,213,0,0" VerticalAlignment="Top" Width="33"/>
+        <Button x:Name="button2" Content="Start Crawling" HorizontalAlignment="Left" Margin="76,315,0,0" VerticalAlignment="Top" Width="92" Height="30"/>
+
```
Comment (olangness) [inline WebCrawler/MainWindow.xaml]: These should not be here. The model should work regardless of view/ui components. These need to be arguments to an api or something. Think of his as you making a web crawling library for C#

File: WebCrawler/MainWindow.xaml.cs [hunk 2]
```diff
@@ -15,14 +28,86 @@
 
 namespace WebCrawler
 {
-    /// <summary>
-    /// Interaction logic for MainWindow.xaml
-    /// </summary>
     public partial class MainWindow : Window
```
Comment (olangness) [inline WebCrawler/MainWindow.xaml.cs:7]: This code should not be in mainwindow class. Models have no knowledge of view. It needs to be separate as a model and not in the codebehind.

File: WebCrawler/obj/Debug/App.g.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿#pragma checksum "..\..\App.xaml" "{8829d00f-11b8-4213-878b-770e8597ac16}" "2ACA586DE23D1095C00C238C6D9C1B1D99DCE0EDCDEE411679567BBD14A99F1F"
+﻿#pragma checksum "..\..\App.xaml" "{ff1816ec-aa5e-4d10-87f7-6f4963833460}" "68BC721C71289E071597F392E11D572D9C8353B2"
```
Comment (olangness) [inline WebCrawler/obj/Debug/App.g.cs]: Delete all the files in in the obj/Debug folder. for some reason the .gitignore isnt working and we dont need these build files. I think the issue will be fixed once #5 is merged

**Review by olangness — CHANGES_REQUESTED** [review]
Make sure your models are separate from the view. The models have no knowledge of the view.

File: WebCrawler/Model/LinkParser.cs [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        /// <summary>
+        /// Exceptions
+        /// </summary>
+        public List<string> Exceptions
+        {
+            get { return _exceptions; }
+            set { _exceptions = value; }
+        }
+
+        #endregion
```
Comment (olangness) [inline WebCrawler/Model/LinkParser.cs]: These can all be [auto implemented properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#auto-implemented-properties). Way nicer

File: WebCrawler/Model/Page.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        public string Url
+        {
+            get { return _url; }
+            set { _url = value; }
+        }
+        #endregion
+    }
+}
```
Comment (olangness) [inline WebCrawler/Model/Page.cs]: Use [Auto Implemented Properties ](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#auto-implemented-properties) to make it nicer and easier to change

**Review by olangness — CHANGES_REQUESTED** [review]

**Comment by olangness** [comment]
Also I dont know if you guys are done or not so If you only want me to review your changes once you are done just close this pr and make one once you are actually done and ready to pull request

**Review by DrewWick — APPROVED** [review]

**Review by DrewWick — DISMISSED** [review]

**Review by olangness — DISMISSED** [review]

**Review by olangness — APPROVED** [review]

**Pull request merged**